src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+
+        /// <summary>
+        /// The number of minutes to wait before evicting an entry from the cache.
+        /// </summary>
+        private const int minutesBeforeRefresh = 60;
+
+        private readonly Queue<string> filePathQueue;
+        private readonly ReaderWriterLockSlim filePathQueueLock;
+
+        /// <summary>
+        /// Handles the actual accessing of data 
+        /// </summary>
+        private readonly DevCanvasAccessor accessor;
+
+        public DevCanvasResultSourceService(
+            string solutionRootPath,
+            Func<string, bool> getOptionStateCallback,
+            IServiceProvider serviceProvider,
+            IHttpClientAdapter httpClientAdapter,
+            ISecretStoreRepository secretStoreRepository,
+            IFileWatcher fileWatcherGitPush,
+            IFileWatcher fileWatcherBranchChange,
+            IFileSystem fileSystem,
+            IGitExe gitExe,
+            IInfoBarService infoBarService,
+            IStatusBarService statusBarService)
+        {
+            this.GetOptionStateCallback = getOptionStateCallback;
+            this.serviceProvider = serviceProvider;
+            this.httpClientAdapter = httpClientAdapter;
+            this.secretStoreRepository = secretStoreRepository;
+            this.fileWatcherBranchChange = fileWatcherBranchChange;
+            this.fileWatcherGitPush = fileWatcherGitPush;
+            this.fileSystem = fileSystem;
+            this.gitExe = gitExe;
+            this.gitExe.RepoPath = solutionRootPath;
+            this.infoBarService = infoBarService;
+            this.statusBarService = statusBarService;
+
+            this.browserService = new BrowserService();
+            this.filesQueriedCache = new MemoryCache("FilesQueried");
+
+            filePathQueue = new Queue<string>();
+
+            accessor = new DevCanvasAccessor();
+        }
+
+        /// <inheritdoc/>
+        public async System.Threading.Tasks.Task InitializeAsync()
+        {
+            string repoPath = await gitExe.GetRepoRootAsync();
+            string repoUrl = await gitExe.GetRepoUriAsync();
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result> IsActiveAsync()
+        {
+            return Result.Success();
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result<bool, ErrorType>> OnDocumentEventAsync(string[] filePaths)
+        {
+            foreach (string filePath in filePaths)
+            {
+                if (!filesQueriedCache.Contains(filePath))
+                {
+                    filesQueriedCache.Add(filePath, DateTime.UtcNow, DateTime.UtcNow.AddMinutes(60));
+                    DownloadInsights(filePath);
+                    await this.statusBarService.SetStatusTextAsync($"Retrieving results from DevCanvas for {filePath}...");
```

Comment:
This message seems more appropriate for the SARIF Viewer output pane rather than the status bar - as it will change fast and could be overwritten often when multipel files are added?

Reply:
If we did have a status bar relevant, it would maybe make sense to have it be "Retrieving results from DevCanvas" when the queue is non-empty?

Reply:
We already print out what file we're getting insights for, I'll remove this for now since I think it would be redundant either way.